src/librustc_resolve/late.rs

```diff
@@ -1843,6 +1898,24 @@ impl<'a, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
         if let Some(LexicalScopeBinding::Res(res)) = binding { res != Res::Err } else { false }
     }
 
+    /// A wrapper around [`Resolver::report_error`].
+    ///
+    /// This doesn't emit errors for function bodies if `ignore_bodies` is set.
+    fn report_error(&self, span: Span, resolution_error: ResolutionError<'_>) {
+        if self.should_report_errs() {
+            self.r.report_error(span, resolution_error);
+        }
+    }
+
+    #[inline]
+    fn should_report_errs(&self) -> bool {
+        debug!("should_report_errs(state={:?})", self.ignore_bodies);
+        match self.ignore_bodies {
+            None | Some(IgnoreState::Report) => true,
```

Comment:
The optional enum setup looks overcomplicated.
`fn should_report_errs` could return `self.sess.opts.actually_rustdoc && self.in_body` where `in_body` is a simple boolean flag.

Reply:
I guess no one will ever want to use this besides rustdoc, yeah. I'll get rid of the `ignore_bodies` parameter to `Visitor::new`, then it will be even more clear this is a hack.